test/extended/router/gatewayapicontroller.go

```
@@ -539,3 +591,74 @@ func assertHttpRouteSuccessful(oc *exutil.CLI, gwName, name string) (*gatewayapi
+	if errGwStatus != nil || gateway == nil {
+		e2e.Failf("Unable to create httpRoute, no gateway available during route assertion %v", errGwStatus)
+	}
+
+	// Create the http client to check the header.
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	// Make sure the DNSRecord is ready to use.
+	assertDNSRecordStatus(oc, gwName, "", "")
+
+	// Wait and check that the dns name resolves first. Takes a long time, so
+	// if the hostname is actually an IP address, skip this.
+	dnsResolutionTimeout := 10 * time.Minute
+	if net.ParseIP(hostname) == nil {
+		if err := wait.PollUntilContextTimeout(context.Background(), 20*time.Second, dnsResolutionTimeout, false, func(context context.Context) (bool, error) {
+			_, err := net.LookupHost(hostname)
+			if err != nil {
+				e2e.Logf("%v waiting for HTTP route name %s to resolve (%v)", time.Now(), hostname, err)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			e2e.Failf("HTTP route name %s was unable to be resolved: %v", hostname, err)
+		}
+	}
+
+	// Wait for http route to respond, and when it does, check for the status code.
+	if err := wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 5*time.Minute, false, func(context context.Context) (bool, error) {
+		statusCode, err := getHttpResponse(client, hostname)
+		if err != nil {
+			e2e.Logf("GET %s failed: %v, retrying...", hostname, err)
+			return false, nil
+		}
+		if statusCode != http.StatusOK {
+			e2e.Logf("GET %s failed: status %v, expected %v, retrying...", hostname, statusCode, http.StatusOK)
+			return false, nil // retry on 503 as pod/service may not be ready
+		}
+		e2e.Logf("request to %s was successful", hostname)
+		return true, nil
+
+	}); err != nil {
+		e2e.Failf("error contacting %s's endpoint: %v", hostname, err)
+	}
+
+	return nil
+}
```

Same as for the other assert function - you don't need to return any error because you use assertions inside the function.
```suggestion
	}
}
```

---

done